src/main/java/mike/Mike.java

```diff
@@ -0,0 +1,106 @@
+        }
+    }
+
+    public String getResponse(String input) {
+        try {
+            int endIndOfCommand = input.indexOf(' ');
+            String descriptionInput = input.substring(endIndOfCommand);
+        } catch (StringIndexOutOfBoundsException e) {
+            try {
+                Command command = Parser.parseCommand(input);
+                switch (command) {
+                case BYE:
+                    return ui.exit();
+                case LIST:
+                    try {
+                        return ui.showList(taskList);
+                    } catch (NullPointerException e2) {
+                        return ui.showListError();
+                    }
+                default:
+                    return ui.showGeneralError();
+                }
+            } catch (IllegalArgumentException e2) {
+                return ui.showGeneralError();
+            }
+        }
+
+        try {
+            int endIndOfCommand = input.indexOf(' ');
+            String commandInput = input.substring(0, endIndOfCommand);
+            String descriptionInput = input.substring(endIndOfCommand + 1);
+            System.out.println(descriptionInput);
+            Command command = Parser.parseCommand(commandInput);
+            switch (command) {
+            case DONE:
+                try {
+                    int i = Integer.parseInt(descriptionInput);
+                    return ui.showMarkSuccess(taskList.mark(i));
+                } catch (IndexOutOfBoundsException e) {
+                    return ui.showIndexOutOfBoundsError(taskList);
+                }
+            case DELETE:
+                try {
+                    int k = Integer.parseInt(descriptionInput);
+                    Task deletedTask = taskList.delete(k);
+                    return ui.showDeleteSuccess(taskList, deletedTask);
+                } catch (IndexOutOfBoundsException e) {
+                    return ui.showIndexOutOfBoundsError(taskList);
+                }
+            case FIND:
+                return ui.showMatchingResults(taskList.find(descriptionInput));
+            default:
+                try {
+                    Task addedTask = Parser.parseDescription(command, descriptionInput);
+                    taskList.add(addedTask);
+                    String response = ui.showAddSuccess(taskList, addedTask);
+                    storage.save(taskList);
+                    return response;
+                } catch (ParseException e) {
+                    return ui.showError(e);
+                } catch (IOException e) {
+                    return ui.showIOError(e);
+                } catch (DateTimeParseException e) {
+                    return ui.showDateTimeParseError(e);
+                }
+            }
+        } catch (IllegalArgumentException e) {
+            return ui.showGeneralError();
+        }
+    }
+
+    public static void main(String[] args) {
+        Mike mike = new Mike("data/tasks.txt");
+    }
+
+}
```

Comment:
Code is really readable and follows Java Coding Standards. Could consider adding header comments to the getResponse() method to improve ease of understanding.